### PR TITLE
(PUP-4603) Use Puppet Server 2.1+ for acceptance tests

### PR DESCRIPTION
**Pull request by MikaelSmith**
- (maint) Update Beaker to latest and pin gems
- (maint) Remove work-around for BKR-262
- (PUP-4603) Use Puppet Server 2.1+

### CODE REVIEW

**Comment by MikaelSmith** [comment]
Should be merged after #26 to make resolving conflicts easier.

**Comment by MikaelSmith** [comment]
Fails to find activemq, pushing an update soon.

File: spec/spec_helper_acceptance.rb [hunk 1]
```diff
@@ -72,7 +68,12 @@ def parser_opts
   {
     :main => {:stringify_facts => false, :parser => 'future', :color => 'ansi'},
     :agent => {:stringify_facts => false, :cfacter => true, :ssldir => '$vardir/ssl'},
-    :master => {:stringify_facts => false, :cfacter => true},
+  }
+end
+
+def server_opts
+  {
+    :master => {:autosign => true},
```
Comment (joshcooper) [inline spec/spec_helper_acceptance.rb]: it doesn't look like `server_opts` isn't used anymore (previously we called `master_opts`)

Comment (MikaelSmith) [inline spec/spec_helper_acceptance.rb]: Oops, I think I mixed that up in a rebase.

**Comment by MikaelSmith** [comment]
Should pass acceptance now.

**Comment by joshcooper** [comment]
It doesn't look like there were code changes, did you force push?

**Pull request merged**